operator/v1/types_ingress.go

```diff
@@ -1346,6 +1346,78 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// For Setting Headers you can provide a value for the header whereas for deleting one needs to provide header name.
+	// +optional
+	Actions IngressControllerHTTPHeaderActions `json:"actions,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActions defines whether we want to set/delete Response or Request Headers.
+type IngressControllerHTTPHeaderActions struct {
+	// Response is a list of http response headers which either need to set or deleted.
+	Response []IngressControllerHTTPHeader `json:"response,omitempty"`
+	// Request is a list of http request headers which either need to set or deleted.
+	Request []IngressControllerHTTPHeader `json:"request,omitempty"`
+}
+
+// IngressControllerSetOrDeleteHTTPHeader specifies options for set/delete a header.
+type IngressControllerHTTPHeader struct {
+	// name specifies a header name to be set/delete.  Its value must be a valid HTTP header.
+	// name as defined in RFC 2616 section 4.2.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	Name string `json:"name"`
+
+	// value specifies fields to mention the header details and the action you want to perform on them.
+	Value IngressControllerHTTPHeaderActionUnion `json:"value"`
+}
+
+// IngressControllerHTTPHeaderActionUnion specifies fields to mention the header details and the action you want to perform on them.
+// +union
+type IngressControllerHTTPHeaderActionUnion struct {
+	// action defines the type of the action to be applied on the header.
+	// Set allows you to set http request and response headers.
+	// Delete allows you to delete http request and response headers
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Set;Delete
+	// +required
+	Action IngressControllerHTTPHeaderActionType `json:"action,omitempty"`
+
+	// set defines HTTP headers that should be set.
+	// If this field is empty, no headers are added.
```

Comment:
Does omitting this mean that no action is taken?
```suggestion
	// set specifies how the HTTP header should be set.
	// If this field is empty, no action is taken.
```

Reply:
yes because we check empty nil conditions